chef/cookbooks/crowbar/templates/default/crowbar-upgrade-os.erb

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# This script starts the upgrade of the node server to the latest version of
+# system and Cloud product.
+
+LOGFILE=/var/log/crowbar/node-upgrade.log
+mkdir -p "`dirname "$LOGFILE"`"
+
+# Copy stdout to fd 3
+exec 3>&1
+# Create fd 4 for logfile
+exec 4>> "$LOGFILE"
+# Set fd 1 and 2 to logfile
+exec 1>&4 2>&1
+
+set -x
+
+initiate_node_upgrade()
+{
+    local upgradedir=/var/lib/crowbar/upgrade
+    mkdir -p $upgradedir
+
+    if [[ -f $upgradedir/node-upgrading ]] ; then
+        echo "Exit: Upgrade is already running..."
+        exit 1
+    fi
+
+    if [[ -f $upgradedir/node-upgraded-ok ]] ; then
```

Comment:
Strictly speaking, I guess you need to check for the content of the file now :-)